pkgs/build-support/dart/pub2nix/package-graph.py

```diff
@@ -0,0 +1,58 @@
+"""
+https://github.com/dart-lang/pub/issues/4522
+This script generates a package_graph.json file.
+"""
+
+import json
+import os
+from pathlib import Path
+from urllib.parse import unquote, urlparse
+
+import yaml
+
+
+def get_package(pubspec_path: Path, dev_dependencies: bool = False):
+    with pubspec_path.open("r", encoding="utf-8") as f:
+        pubspec = yaml.safe_load(f)
+    package = {
+        "name": pubspec["name"],
+        "version": pubspec.get("version") or "0.0.0",
+        "dependencies": list((pubspec.get("dependencies") or {}).keys()),
+    }
+    if dev_dependencies:
+        package["devDependencies"] = list(
+            (pubspec.get("dev_dependencies") or {}).keys()
+        )
```

Comment:
```suggestion
    package = {
        "name": pubspec["name"],
        "version": pubspec.get("version") or "0.0.0",
        "dependencies": list(pubspec.get("dependencies") or {}),
    }
    if dev_dependencies:
        package["devDependencies"] = list(pubspec.get("dev_dependencies") or {})
```

Equivalent and slightly faster